.github/workflows/proxy_plugin.yml

```diff
@@ -0,0 +1,68 @@
+---
+name: Foreman Proxy Plugin
+
+on:
+  workflow_call:
+    inputs:
+      foreman_proxy_version:
+        description: Foraman proxy version to use
+        default: 'develop'
+        required: false
+        type: string
+      foreman_proxy_repository:
+        description: Foreman proxy repository to use
+        default: 'theforeman/smart-proxy'
+        required: false
+        type: string
+      matrix_include:
+        description: Additional test matrix combinations to include
+        default: '[]'
+        required: false
+        type: string
+      matrix_exclude:
+        description: Test matrix combinations to exclude
+        default: '[]'
+        required: false
+        type: string
+
+env:
+  BUNDLE_WITHOUT: "bmc:development:dhcp_isc_inotify:dhcp_isc_kqueue:journald:krb5:libvirt:puppetca_token_whitelisting:realm_freeipa:windows"
```

Comment:
I've been thinking about making the gemfiles smarter. Like using `RUBY_PLATFORM` to detect if it's BSD or Linux. Something like https://github.com/theforeman/smart-proxy/pull/884.

And I'm not sure we should be so aggressive in disabling all the modules. Or we can provide an env var that toggles `optional` in the groups.